lib/pharos/phase_manager.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Pharos
+  class PhaseManager
+    include Pharos::Logging
+
+    # @param dirs [Array<String>]
+    def initialize(dirs, ssh_manager: , **options)
+      @ssh_manager = ssh_manager
+      @options = options
+
+      load_phases(dirs)
+    end
+
+    # @param phases [Array<Pharos::Phases::Base>]
+    # @return [Array<...>]
+    def run_parallel(phases)
+      threads = phases.map { |phase|
+        Thread.new do
```

Comment:
This is now an entirely generic parallel map, with nothing at all specific to phases.